### PR TITLE
Ref task fix timeline

### PR DESCRIPTION
**Pull request by DoanVanToan**

### CODE REVIEW

File: app/src/main/java/com/framgia/englishconversation/screen/audiodetail/AudioDetailViewModel.java [hunk 1]
```diff
@@ -144,7 +146,7 @@ public TimelineModel getTimelineModel() {
         return mTimelineModel;
     }
 
-    private class ComponentListener implements Player.EventListener {
+    private class ComponentListener implements ExoPlayer.EventListener {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/audiodetail/AudioDetailViewModel.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
+
 import com.framgia.englishconversation.data.model.MediaModel;
 import com.framgia.englishconversation.data.model.TimelineModel;
 import com.framgia.englishconversation.databinding.ItemTimelineAudioBinding;
 import com.framgia.englishconversation.databinding.ItemTimelineConversationBinding;
 import com.framgia.englishconversation.databinding.ItemTimelineImageBinding;
 import com.framgia.englishconversation.databinding.ItemTimelineOnlyTextBinding;
 import com.framgia.englishconversation.databinding.ItemTimelineVideoBinding;
+
+import java.net.URL;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: ERROR: Unused import: java.net.URL. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'java.net.URL'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/BaseMediaTimelineViewHolder.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    public void release() {
+        if (mHelper != null) {
+            mHelper.release();
+            mHelper = null;
+        }
+    }
+
+    @Override
+    public boolean wantsToPlay() {
+        return ToroUtil.visibleAreaOffset(this, itemView.getParent()) >= VISIBLE_ARE_TIMELINE_ADAPTER;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/BaseMediaTimelineViewHolder.java]: ERROR: Line is longer than 100 characters (found 102).

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java [hunk 4]
```diff
@@ -1,23 +1,36 @@
 package com.framgia.englishconversation.screen.timeline;
 
+import android.net.Uri;
+import android.support.annotation.NonNull;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: ERROR: Unused import: android.support.annotation.NonNull. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java [hunk 5]
```diff
@@ -1,23 +1,36 @@
...
+import android.support.annotation.Nullable;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: ERROR: Unused import: android.support.annotation.Nullable. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java [hunk 6]
```diff
@@ -1,23 +1,36 @@
...
 import com.framgia.englishconversation.data.model.MediaModel;
 import com.framgia.englishconversation.data.model.TimelineModel;
 import com.framgia.englishconversation.databinding.ItemTimelineAudioBinding;
 import com.framgia.englishconversation.databinding.ItemTimelineConversationBinding;
 import com.framgia.englishconversation.databinding.ItemTimelineImageBinding;
 import com.framgia.englishconversation.databinding.ItemTimelineOnlyTextBinding;
 import com.framgia.englishconversation.databinding.ItemTimelineVideoBinding;
+import com.google.android.exoplayer2.ui.SimpleExoPlayerView;
+
+import java.net.URL;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: ERROR: Unused import: java.net.URL. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java [hunk 7]
```diff
@@ -1,23 +1,36 @@
...
 import java.util.List;
 
+import im.ene.toro.ToroPlayer;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: ERROR: Unused import: im.ene.toro.ToroPlayer. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java [hunk 8]
```diff
@@ -1,23 +1,36 @@
...
+import im.ene.toro.ToroUtil;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: ERROR: Unused import: im.ene.toro.ToroUtil. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java [hunk 9]
```diff
@@ -1,23 +1,36 @@
...
+import im.ene.toro.exoplayer.SimpleExoPlayerViewHelper;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: ERROR: Unused import: im.ene.toro.exoplayer.SimpleExoPlayerViewHelper. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java [hunk 10]
```diff
@@ -1,23 +1,36 @@
...
+import im.ene.toro.media.PlaybackInfo;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: ERROR: Unused import: im.ene.toro.media.PlaybackInfo. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java [hunk 11]
```diff
@@ -1,23 +1,36 @@
...
+import im.ene.toro.widget.Container;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: ERROR: Unused import: im.ene.toro.widget.Container. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java (hunk 4 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.support.annotation.NonNull'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java (hunk 5 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.support.annotation.Nullable'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java (hunk 6 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'java.net.URL'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java (hunk 7 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'im.ene.toro.ToroPlayer'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java (hunk 8 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'im.ene.toro.ToroUtil'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java (hunk 9 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'im.ene.toro.exoplayer.SimpleExoPlayerViewHelper'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java (hunk 10 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'im.ene.toro.media.PlaybackInfo'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java (hunk 11 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/timeline/TimelineAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'im.ene.toro.widget.Container'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

**Review by anhpham2403 — APPROVED** [review]

**Pull request merged**